### PR TITLE
feat(ESSNTL-3716): Add Inventory groups column and filter

### PR DESCRIPTION
**Pull request by Fewwy**
# Description
Add group filter and column hidden behind feature flag
Associated Jira ticket: # (issue)
https://issues.redhat.com/browse/ESSNTL-3716
Please include a summary of the change, what this fixes/creates/improves.

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Review by leSamo — CHANGES_REQUESTED** [review]
When I try to disable feature flag:
Screenshot from 2023-07-18 12-01-58

When I try to filter:
[Screencast from 2023-07-18 11-59-08.webm](https://github.com/RedHatInsights/patchman-ui/assets/8426204/4c3a7152-a1d0-4ab5-8388-66e91c65275c)

File: src/Utilities/SystemsHelpers.js [hunk 1]
```diff
@@ -49,10 +50,24 @@ export const buildActiveFiltersConfig = (filter, search, deleteFilters) => ({
...
+    const groupColumn = defaultColumns.filter(({ key }) => key === 'groups');
+    const tagsColumn = defaultColumns.filter(({ key }) => key === 'tags');
 
-    let nameAndTag = defaultColumns.filter(({ key }) => key === 'display_name' || key === 'tags');
+    groupColumn[0]?.renderFunc ? groupColumn[0].renderFunc = (data, ...value) => {
+        const groups = value[1].attributes?.groups;
+        return isEmpty(groups) ? (
+            'N/A'
+        ) : (
+            //change this link to a InsightsLink after router upgrade
```
Comment (gkarat) [inline src/Utilities/SystemsHelpers.js:62]:
```suggestion
            // TODO: change this link to a InsightsLink after router upgrade
```

**Review by gkarat — CHANGES_REQUESTED** [review]
When I apply the group filter by choosing a group that has some hosts, I see that data is not filtered. However, the backend already has this filter implemented https://issues.redhat.com/browse/ESSNTL-4817?focusedId=22636009&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22636009. I also tried to edit and resend the request (to /api/patch/v3/system) with this parameter and it indeed filtered hosts. So let's support it here too within this PR.

The group column is mapped properly and for some hosts I see the link is rendered and it works correct ✅

**Pull request closed without merging**